Sources/StreamChatUI/CommonViews/AvatarView/ChatChannelAvatarView.swift

```diff
@@ -30,31 +47,293 @@ open class ChatChannelAvatarView: _View, ThemeProvider, SwiftUIRepresentable {
+            
+            self.presenceAvatarView.avatarView.imageView.loadImage(
+                from: nil,
+                placeholder: combinedImage,
+                preferredSize: .avatarThumbnailSize,
+                components: self.components
+            )
+        }
+    }
+    
+    /// Loads avatars for the given URLs
+    /// - Parameters:
+    ///   - urls: The avatar urls
+    ///   - channelId: The channelId of the channel
+    ///   - completion: Completion that gets called with an array of `UIImage`s when all the avatars are loaded
+    open func loadAvatarsFrom(
+        urls: [URL?],
+        channelId: ChannelId,
+        completion: @escaping ([UIImage], ChannelId)
+            -> Void
+    ) {
+        var placeholderAvatars: [UIImage] = []
+        
+        var placeholderImages = [
+            appearance.images.userAvatarPlaceholder1,
+            appearance.images.userAvatarPlaceholder2,
+            appearance.images.userAvatarPlaceholder3,
+            appearance.images.userAvatarPlaceholder4
+        ]
+        
+        var avatarUrls: [URL] = []
+        
+        for url in urls.prefix(4) {
+            if let avatarUrl = url {
+                avatarUrls.append(avatarUrl)
+            } else {
+                placeholderAvatars.append(placeholderImages.removeFirst())
+            }
+        }
+        
+        let group = DispatchGroup()
+        var images: [UIImage] = []
+        
+        for avatarUrl in avatarUrls {
+            var placeholderIndex = 0
+            group.enter()
+            
+            let thumbnailUrl = components.imageCDN.thumbnailURL(originalURL: avatarUrl, preferredSize: .avatarThumbnailSize)
+            let imageRequest = components.imageCDN.urlRequest(forImage: thumbnailUrl)
+            let cachingKey = components.imageCDN.cachingKey(forImage: avatarUrl)
+
+            imageLoader.loadImage(using: imageRequest, cachingKey: cachingKey) { result in
+                switch result {
+                case let .success(image):
+                    images.append(image)
+                case .failure:
+                    if !placeholderImages.isEmpty {
+                        // Rotationally use the placeholders
+                        images.append(placeholderImages[placeholderIndex])
+                        placeholderIndex += 1
+                        if placeholderIndex == placeholderImages.count {
+                            placeholderIndex = 0
+                        }
+                    }
+                }
+                group.leave()
+            }
+            
+            group.notify(queue: .main) {
+                completion(images, channelId)
+            }
+        }
+    }
+    
+    /// Creates a merged avatar from the given images
+    /// - Parameter avatars: The individual avatars
+    /// - Returns: The merged avatar
+    open func createMergedAvatar(from avatars: [UIImage]) -> UIImage? {
```

Comment:
for me this name is more like `createMergedAvatarOrPlaceholder`, but please think on a better naming

cause for caller its unobvious that in case of !avatars.isEmpty it will return placeholder